### PR TITLE
offline onboarding fixes

### PR DESCRIPTION
**Pull request by poocart**
Part of onboarding refactor, this includes:
- Extra fixes for connectivity check.
- Address error on offline retry.
- Multiple offline toast fix.

### CODE REVIEW

File: src/actions/onboardingActions.js [hunk 1]
```diff
@@ -385,6 +384,7 @@ export const checkUsernameAvailabilityAction = (username: string) => {
 
     // if user is offline then proceed with local registration
     if (!getState()?.session?.data?.isOnline) {
+      Toast.closeAll(); // close current ones to not block with multiple toasts
```
Comment (graszka22) [inline src/actions/onboardingActions.js:387]: 🤔  I'm not getting multiple modals anymore 👍  but they're jumping every time I enter a letter and it's still covering input a bit. Can we show this toast just once? If that's hard, let's leave as it is.

Screenshot 2020-10-07 at 16 46 22

File: src/actions/historyActions.js [hunk 2]
```diff
@@ -122,6 +123,8 @@ export const fetchSmartWalletTransactionsAction = () => {
       }
     }
 
+    if (!isOnline) return;
```
Comment (graszka22) [inline src/actions/historyActions.js:126]: Are those checks necessary? I've counted 4 of them in this PR :p Should we remember about offline check every time we make an api call?

**Review by graszka22 — COMMENTED** [review]
I have two suggestions and two bugs:
1.
- create account while offline
- be home screen
- close the app
- turn on the internet
- start the app, enter pin, registration failed

2. Sign out while being offline. A crash with "Error: failed" message should appear

File: src/actions/historyActions.js (hunk 2 above)
Comment (poocart) [reply src/actions/historyActions.js:126]: Per out last discussion this is needed, however, we can think how we can improve the check throughout the app, however, it's not matching this ticket scope.

File: src/actions/onboardingActions.js (hunk 1 above)
Comment (poocart) [reply src/actions/onboardingActions.js:387]: Thanks, solved

**Review by IAmKio — APPROVED** [review]
Great work mate

**Review by graszka22 — APPROVED** [review]

**Pull request merged**